### DISCUSSION TITLE
Alternative to outer_type_ in Pydantic v2

### DISCUSSION THREAD

**Discussion by deivydaslipskis** [pydantic/pydantic#7217] [Pydantic V2]
What would be an alternative to `outer_type_` in Pydantic v2 since it is no longer available?

I want to check if the type of model is a `List`.

Example of how I used it in Pydantic v1:

```
print(value)
ModelField(name='employees', type=List[Person], required=true)
if get_origin(value.outer_type_) == list:
    # do something
```

As I can see, now instead of `ModelField` there is `FieldInfo`.

```
print(value)
FieldInfo(annotation=List[Person], required=True)
```

And now `value` no longer has `outer_type_` method which would help me identify if it is a `List`.

**Comment by harunyasar** [comment]
```python
field_info = FieldInfo(annotation=List[Person], required=True)
assert field_info.annotation.__origin__ == list
```

**Reply by deivydaslipskis** [reply]
Thanks!

**Comment by DLogunoff** [comment]
Is this possible to replace v1 pydantic's FieldInfo.outer_type_ if I need to use it this way

```Python
class SomeModel:
    field: Optional[bool]

SomeModel.__fields__['field'].outer_type_ == bool
```

Solution provided above gives me:

```Python
typing.Union
```

**Reply by ph-ausseil** [reply]
@DLogunoff : I think 
```python 
SomeModel.__fields__['field'].annotation.__args__[0] 
```

May give you what you want 🤢
